src/v/model/transform.h

```diff
@@ -102,14 +102,20 @@ struct transform_offset_options
     //
     // When a timestamp is used, a timequery is used to resolve the offset for
     // each partition.
-    serde::variant<latest_offset, model::timestamp> position;
+    //
+    // When an offset_delta is used, offset resolution depends on its sign:
+    //   - if delta < 0  - start at latest - abs(delta) (for each partition)
+    //   - if delta >= 0 - start at earliest + delta (for each partition)
```

Comment:
Not to be a pain but - we could get rid of the awkwardness in rpk with -0 if we represented things more similar to the API (which is more verbose and leaning into the sumtype more)

by making latest_offset struct have a delta field meaning the delta from the end, and adding a new struct for earliest offset and adding a delta field there too.

Reply:
We could translate latest_offset delta=0 to freeze offset like we do now, or we could leave latest offset and just have two new variants for from_beginning and from_end with (positive) deltas

Reply:
Not a pain...I thought about doing that, but it seemed a bit over complex at the time. I'm not opposed. You could argue that the current approach just pushes the complexity into code comments and across the api boundary anyway, which is not great.